### ISSUE TITLE
Unable to compile on current nightly with Python 3.7

### ISSUE THREAD

**Issue by damienstanton** [PyO3/pyo3#201]
## 🐛 Bug Reports

When reporting a bug, please provide the following information. If this is not a bug report you can just discard this template.

### 🌍 Environment

 - Your operating system and version: macOS 10.13.6
 - Your python version: 3.7.0
 - How did you install python (e.g. apt or pyenv)? Did you use a virtualenv?: Brew installed Python, no virtualenv (also tried with pyvenv -m venv, same results)
 - Your rust version (`rustc --version`): rustc 1.30.0-nightly (73c78734b 2018-08-05)

### 💥 Reproducing

Please provide a [minimal working example](https://stackoverflow.com/help/mcve). This means both the rust code and the python.

```python
# setup.py
'''
a simple FFI example of a Rust crate exported as a native Python module
'''
import sys
from setuptools import setup

try:
    from setuptools_rust import Binding, RustExtension

except ImportError:
    import subprocess
    ERR = subprocess.call(
        [sys.executable, '-m', 'pip3', 'install', 'setuptools-rust'])
    if ERR:
        print("setuptools-rust not installed")
        raise SystemExit(ERR)
    else:
        from setuptools_rust import Binding, RustExtension

SETUP_REQUIRES = ['setuptools-rust>=0.9.2']
# Note: install_requires would be packages internal to the python module being exposed
INSTALL_REQUIRES = []

setup(
    name='lenrs',
    version='0.1',
    classifiers=[
        'License :: OSI Approved :: MIT License',
        'Development Status :: 3 - Alpha', 'Intended Audience :: Developers',
        'Programming Language :: Python', 'Programming Language :: Rust',
        'Operating System :: POSIX', 'Operating System :: MacOS :: MacOS X'
    ],
    rust_extensions=[
        RustExtension('lenrs._lenrs', 'Cargo.toml', binding=Binding.PyO3)
    ],
    packages=['lenrs'],
    zip_safe=False)

```

```python
# __init.py
from ._lenrs import len
```

```rust
// lib.rs
#![feature(proc_macro)]
#![feature(proc_macro_path_invoc)]

#[macro_use]
extern crate pyo3;
use pyo3::prelude::*;

#[py::modinit(_lenrs)]
fn init(py: Python, m: &PyModule) -> PyResult<()> {
    #[pyfn(m, "len")]
    fn len(py: Python, s: &str) -> PyResult<PyObject> {
        Ok(s.len().to_object(py))
    }

    Ok(())
}
```

```toml
# Cargo.toml
```

Please also write what exact flags are required to reproduce your results.

I have `setuptools-rust` installed via pip3, so I use no extra flags.

There are 11 implementation errors such as this:
```shell
error[E0119]: conflicting implementations of trait `python::IntoPyDictPointer` for type `(_, _)`:
   --> /Users/damien/.cargo/registry/src/github.com-1ecc6299db9ec823/pyo3-0.3.2/src/objects/dict.rs:246:1
    |
231 | / impl <K, V, I> IntoPyDictPointer for I
232 | | where
233 | |     K: ToPyObject,
234 | |     V: ToPyObject,
...   |
243 | |     }
244 | | }
    | |_- first implementation here
```

The full error output can be found [at this gist link](https://gist.github.com/damienstanton/c315d395b7e027aec8407107cc8bf0c0)

**Comment by konstin** [comment]
That's fixed in 0.4.0. Your rust code is also not compatible with the current nightly since `proc_macro` has been replaced with `use_extern_macros`; There's more on that in the changelog.

**Comment by damienstanton** [comment]
Interesting, okay. I think the tutorial I was following was out of date. If I make these changes and match the macros with the example in the readme, all looks good to go. I'll let the tutorial author know it doesn't work on their end as described. Thanks!
